Classes/Console/Command/BackendCommandController.php

```diff
@@ -142,7 +142,7 @@ public function unlockForEditorsCommand()
     public function createAdminCommand(string $username, string $password)
     {
         $givenUsername = $username;
-        $username = preg_replace('/[^a-z0-9]/', '', $username);
+        $username = preg_replace('/[^a-z0-9-_]/', '', $username);
```

Comment:
Good point, that this regex it too restrictive.
Can you implement something that matches the default in TYPO3, which is:
`'eval' => 'nospace,trim,lower,unique,required'`

This means every character is allowed, but no spaces and everything will be lowercased.

Install tool admin creation has this: `$username = preg_replace('/\\s/i', '', $request->getParsedBody()['install']['userName']);`

Reply:
I'd additionally add a strtolower here as well